### PR TITLE
feature(lyz): Add a CarRacing env.

### PR DESCRIPTION
**Pull request by Meta-YZ**
## Description
Add a CarRacing-v0 env in the dizoo/box2d folder.

## TODO

- [ ] Add td3+bc algorithm.

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: dizoo/box2d/car_racing/entry/my.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import gym
```
Comment (PaParaZz1) [inline dizoo/box2d/car_racing/entry/my.py:1]: remove your own test code in committed codes

File: dizoo/box2d/car_racing/config/car_racing_ppo_config.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+from easydict import EasyDict
```
Comment (PaParaZz1) [inline dizoo/box2d/car_racing/config/car_racing_ppo_config.py:1]: modify pull request name following this format ([link](https://di-engine-docs.readthedocs.io/zh_CN/latest/guide/warmup_zh.html#git-commit))

File: dizoo/box2d/car_racing/envs/car_racing_env.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    def _make_env(self):
+        return wrap_deepmind(
+            self._cfg.env_id,
+            frame_stack=self._cfg.frame_stack,
+            # episode_life=self._cfg.is_train,
+            clip_rewards=self._cfg.is_train
+        )
+
+    def __repr__(self) -> str:
+        return "DI-engine Atari Env({})".format(self._cfg.env_id)
```
Comment (PaParaZz1) [inline dizoo/box2d/car_racing/envs/car_racing_env.py:83]: modify name, not Atari

File: dizoo/box2d/car_racing/envs/car_racing_env_wrapper.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# Borrow a lot from openai baselines:
+# https://github.com/openai/baselines/blob/master/baselines/common/atari_wrappers.py
+
+import gym
+from collections import deque
+from ding.envs import NoopResetWrapper, MaxAndSkipWrapper, EpisodicLifeWrapper, FireResetWrapper, WarpFrameWrapper, ScaledFloatFrameWrapper, \
+                        ClipRewardWrapper, FrameStackWrapper
+
+
+def wrap_deepmind(env_id, episode_life=True, clip_rewards=True, frame_stack=4, scale=True, warp_frame=True):
```
Comment (PaParaZz1) [inline dizoo/box2d/car_racing/envs/car_racing_env_wrapper.py:10]: you can refer to this [repo](link) for more proper env wrapper for car racing. Don't need follow atari wrapper strictly.

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: dizoo/box2d/car_racing/envs/test_car_racing.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            random_action = env.action_space.sample()
+            print(random_action)
+            timestep = env.step(random_action)
+            print(timestep)
+            assert isinstance(timestep.obs, np.ndarray)
+            assert isinstance(timestep.done, bool)
+            assert timestep.reward.shape == (1,)
+            assert timestep.reward >= env.reward_space.low
+            assert timestep.reward <= env.reward_space.high
+            # assert isinstance(timestep, tuple)
```
Comment (zxzzz0) [inline dizoo/box2d/car_racing/envs/test_car_racing.py:32]: Does `env.step()` return `tuple` or not?

Comment (PaParaZz1) [reply dizoo/box2d/car_racing/envs/test_car_racing.py:32]: namedtuple, subclass of tuple

File: dizoo/box2d/car_racing/entry/my.py (hunk 1 above)
Comment (Meta-YZ) [reply dizoo/box2d/car_racing/entry/my.py:1]: Deleted.

File: dizoo/box2d/car_racing/envs/car_racing_env.py (hunk 3 above)
Comment (Meta-YZ) [reply dizoo/box2d/car_racing/envs/car_racing_env.py:83]: OK.

File: dizoo/box2d/car_racing/envs/car_racing_env_wrapper.py (hunk 4 above)
Comment (Meta-YZ) [reply dizoo/box2d/car_racing/envs/car_racing_env_wrapper.py:10]: > you can refer to this [repo](link) for more proper env wrapper for car racing. Don't need follow atari wrapper strictly.

Sorry, I can't open the link. Could you please resend it?

**Pull request closed without merging**